src/app/features/home/details/details-iframe/details-iframe.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, Input } from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+import { UntilDestroy } from '@ngneat/until-destroy';
+import { combineLatest, fromEvent, ReplaySubject } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+import {
+  CachedQueryJWTToken,
+  DiaBackendAuthService,
+} from '../../../../shared/dia-backend/auth/dia-backend-auth.service';
+import { BUBBLE_IFRAME_URL } from '../../../../shared/dia-backend/secret';
+import { NetworkService } from '../../../../shared/network/network.service';
+import { DetailedCapture } from '../information/session/information-session.service';
+
+@UntilDestroy()
+@Component({
+  selector: 'app-detailed-capture-iframe',
+  templateUrl: './details-iframe.component.html',
+  styleUrls: ['./details-iframe.component.scss'],
+})
+export class DetailsIframeComponent {
+  readonly networkConnected$ = this.networkService.connected$;
+
+  private readonly detailedCapture$ = new ReplaySubject<DetailedCapture>(1);
+  @Input() set detailedCapture(value: DetailedCapture | undefined) {
+    if (value) this.detailedCapture$.next(value);
+  }
+
+  readonly iframeUrl$ = combineLatest([
+    this.detailedCapture$,
+    this.diaBackendAuthService.cachedQueryJWTToken$,
+  ]).pipe(
+    map(([detailedCapture, token]) => {
+      return this.generateIframeUrl(detailedCapture, token);
+    })
+  );
+
+  readonly iframeLoaded$ = fromEvent(window, 'message').pipe(
+    map(event => (event as MessageEvent).data),
+    filter(data => data === 'iframe-on-load') // TODO: use enum
+  );
+
+  constructor(
+    private readonly diaBackendAuthService: DiaBackendAuthService,
+    private readonly sanitizer: DomSanitizer,
+    private readonly networkService: NetworkService
+  ) {}
+
+  private generateIframeUrl(
+    detailedCapture: DetailedCapture,
+    token: CachedQueryJWTToken
+  ) {
+    const params =
+      `nid=${detailedCapture.id}` +
+      `&token=${token.access}` +
+      `&refresh_token=${token.refresh}` +
+      `&from=mycapture`;
+    const url = `${BUBBLE_IFRAME_URL}/asset_page?${params}`;
+    return this.sanitizer.bypassSecurityTrustResourceUrl(url);
+  }
+}
```

Comment:
ChatGPT Plus code review with improvement suggestions:

1.  Add comments: Add comments to the `DetailsIframeComponent` class, its properties, and methods to provide context and explain their purpose. This will help improve readability and maintainability.

2.  Enum for event filtering: Instead of using a string for event filtering, consider using an enum for better maintainability and to avoid potential typos.

Here's the updated version of the code with the suggestions applied:

```typescript

// Add an enum for iframe events
enum IframeEvent {
  IFrameOnLoad = 'iframe-on-load',
}

@UntilDestroy()
@Component({
  selector: 'app-detailed-capture-iframe',
  templateUrl: './details-iframe.component.html',
  styleUrls: ['./details-iframe.component.scss'],
})
export class DetailsIframeComponent {
  // Other properties...

  // Filter event data based on the enum value
  readonly iframeLoaded$ = fromEvent(window, 'message').pipe(
    map(event => (event as MessageEvent).data),
    filter(data => data === IframeEvent.IFrameOnLoad)
  );

  // Other methods...
}```

1.  Consider using a `const` instead of `let`: In the `generateIframeUrl` method, you are using a `let` variable called `params`. However, you're not reassigning its value. Consider using a `const` variable instead.

With these suggestions, your code will be easier to read, maintain, and less prone to errors.

Reply:
@tammyyang nice to have. Asana [card](https://app.asana.com/0/1201016280880500/1204235837449459/f) created.